lib/iris/pandas.py

```diff
@@ -518,88 +518,366 @@ def _get_base(array):
+            outlist += [[coord.name(), n, _as_pandas_coord(coord)]]
+    return list(chain.from_iterable([outlist]))
+
+
+def _make_ancillary_variables_list(cube):
+    outlist = []
+    # Get Ancillary variables
+    for ancil_var in cube.ancillary_variables():
+        outlist += [
+            [
+                ancil_var.name(),
+                cube.ancillary_variable_dims(ancil_var),
+                ancil_var.data,
+            ]
+        ]
+    return list(chain.from_iterable([outlist]))
+
+
+def _make_cell_measures_list(cube):
+    outlist = []
+    # Get cell measures
+    for cell_measure in cube.cell_measures():
+        outlist += [
+            [
+                cell_measure.name(),
+                cube.cell_measure_dims(cell_measure),
+                cell_measure.data,
+            ]
+        ]
+    return list(chain.from_iterable([outlist]))
+
 
-    Kwargs:
+def as_series(cube, copy=True):
+    """
+    Convert a 1D cube to a Pandas Series.
 
-        * copy - Whether to make a copy of the data.
-                 Defaults to True. Must be True for masked data.
+    .. deprecated:: 3.4.0
+        This function is scheduled for removal in a future release, being
+        replaced by :func:`iris.pandas.as_data_frame`, which offers improved
+        multi dimension handling.
 
-    .. note::
+    Parameters
+    ----------
+    cube: :class:`Cube`
+        The cube to convert to a Pandas Series.
+    copy : bool, default=True
+        Whether to make a copy of the data.
+        Defaults to True. Must be True for masked data.
 
-        This function will copy your data by default.
-        If you have a large array that cannot be copied,
-        make sure it is not masked and use copy=False.
+    Notes
+    -----
+    This function will copy your data by default.
+    If you have a large array that cannot be copied,
+    make sure it is not masked and use copy=False.
 
     """
+    message = (
+        "iris.pandas.as_series has been deprecated, and will be removed in a "
+        "future release. Please use iris.pandas.as_data_frame instead."
+    )
+    warn_deprecated(message)
+
     data = cube.data
     if ma.isMaskedArray(data):
         if not copy:
             raise ValueError("Masked arrays must always be copied.")
         data = data.astype("f").filled(np.nan)
     elif copy:
         data = data.copy()
-
     index = None
     if cube.dim_coords:
         index = _as_pandas_coord(cube.dim_coords[0])
-
     series = pandas.Series(data, index)
     if not copy:
         _assert_shared(data, series)
-
     return series
 
 
-def as_data_frame(cube, copy=True):
+def as_data_frame(
+    cube,
+    copy=True,
+    add_aux_coord=False,
+    add_cell_measures=False,
+    add_ancillary_variables=None,
+):
     """
-    Convert a 2D cube to a Pandas DataFrame.
-
-    Args:
+    :attr:`~iris.cube.Cube.dim_coords` and :attr:`~iris.cube.Cube.data` are flattened into a long-style
```

Comment:
Could you reduce some of the docstring line lengths? We try and stay within `80`, with some leeway when necessary (`~88`).

### Exceptions

- The one-line descriptor at the top of the docstring.
- Examples (which would render terribly otherwise).